### PR TITLE
Added support for user home variable in settings

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
eg `"cursorless.private.hatShapesDir": "C:\\Users\\Andreas\\repositories\\pokey-dotfiles\\cursorless-hat-shapes",`

## Checklist

- [/] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [/] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [/] I have not broken the cheatsheet

### CODE REVIEW

File: packages/cursorless-vscode/src/ide/vscode/VscodeConfiguration.ts [hunk 1]
```diff
@@ -42,3 +46,20 @@ export default class VscodeConfiguration implements Configuration {
 
   onDidChangeConfiguration = this.notifier.registerListener;
 }
+
+export function vscodeGetConfigurationString(path: string): string | undefined {
+  const value = vscode.workspace.getConfiguration().get<string>(path);
```
Comment (pokey) [inline packages/cursorless-vscode/src/ide/vscode/VscodeConfiguration.ts:61]: This loads the whole configuration, right? Before we just loaded the section we cared about. Probably prematurely optimising by caring about this stuff, but I'm assuming there's a reason VSCode supports retrieving just one section?

**Review by pokey — COMMENTED** [review]
Mostly looks good, with minor comment; see also my commit for a couple changes I made

File: packages/cursorless-vscode/src/ide/vscode/VscodeConfiguration.ts (hunk 1 above)
Comment (AndreasArvidsson) [reply packages/cursorless-vscode/src/ide/vscode/VscodeConfiguration.ts:61]: I'm not really sure why vscode has this section separation. If you like I can split on the last dot and just fetch the section first?

Comment (pokey) [reply packages/cursorless-vscode/src/ide/vscode/VscodeConfiguration.ts:61]: Seems reasonable for now

Comment (AndreasArvidsson) [reply packages/cursorless-vscode/src/ide/vscode/VscodeConfiguration.ts:61]: done

**Comment by AndreasArvidsson** [comment]
@pokey poke!

**Review by pokey — APPROVED** [review]

**Pull request merged**